buildSrc/src/main/groovy/org/apache/beam/gradle/BeamDockerPlugin.groovy

```diff
@@ -267,7 +272,19 @@ class BeamDockerPlugin implements Plugin<Project> {
     if (ext.pull) {
       buildCommandLine.add '--pull'
     }
-    buildCommandLine.addAll(['-t', "${-> ext.name}", '.'])
+    if (!ext.tags.isEmpty() && ext.push) {
```

Comment:
I see, it's because the need to attach with multiple tag names

Reply:
Yeah, I tried doing that initially, but found it was impossible to do a push with multiple tags without saving the image first within the current plugin setup (which doesn't work today with docker buildx). We could push the image with a single tag and then have a follow up step to add tags, but I don't think that is worth it. Eventually this will likely be supported and we can undo this change. We could also consider upstreaming this change to the original plugin if needed.

There is also no way to avoid the provenance piece as of today (which in fairness is more annoying than anything, we could keep that if we needed to).

For now, though, I'm inclined to take this because I think it is the cleanest way to accomplish our goals.